crates/swc_ecma_transforms_compat/src/es2022/class_properties/mod.rs

```diff
@@ -516,33 +523,10 @@ impl ClassProperties {
                         });
                     }
 
-                    let callee = helper!(define_property, "defineProperty");
-
                     if prop.is_static {
-                        extra_stmts.push(
-                            CallExpr {
-                                span: DUMMY_SP,
-                                callee,
-                                args: vec![
-                                    class_ident.clone().as_arg(),
-                                    key.as_arg(),
-                                    value.as_arg(),
-                                ],
-                                type_args: Default::default(),
-                            }
-                            .into_stmt(),
-                        )
+                        extra_inits.insert(key, MemberInit::Prop(prop_span, value));
```

Comment:
Is the type of `key` `Expr?

Reply:
I think it's generally not safe to use it as a hash key.

Reply:
This is indeed problemtic, but since public proerties name could be str/num/bigint, I don't find better way.

Reply:
Yeah, I had thought about it and I concluded that it's fine because we parse `NaN` as `Expr::Ident`.
So this is not relevant at the moment of this transform is applied.

Reply:
We also need to preserve evaluation order of member init